paywall/src/data-iframe/postOffice.js

```diff
@@ -71,8 +72,24 @@ export default function postOffice(window, requiredConfirmations) {
   }
 
   const lockHandler = cachedData => {
-    actions.locks(cachedData.locks)
-    const unlockedLockAddresses = Object.values(cachedData.locks)
+    const relevantLocks = getRelevantLocks()
+    if (!relevantLocks) {
+      // this should not happen
```

Comment:
If this may be the case why are we handling this scenario? What are some of the conditions that would cause this branch to be executed?

Reply:
this is a boundary check. If `POST_MESSAGE_SEND_UPDATES` with `'locks'` is sent prior to receiving `POST_MESSAGE_CONFIG` then this will trigger. This can only happen if someone introduces a bug upstream, but it will be exceedingly difficult to debug in that case. This is here to make it easier to debug. I'd like to keep it, as it does no damage, but will allow future us to debug this, should it ever occur. I'm happy to add my explanation in a comment in the source

Reply:
comment added to source